client/components/badge/index.jsx

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { execSync } from 'child_process';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+
+function getCurrentBranchName() {
+	try {
+		return execSync( 'git rev-parse --abbrev-ref HEAD' ).toString().replace( /\s/gm, '' );
+	} catch ( err ) {
+		return undefined;
+	}
+}
+
+function getCurrentCommitShortChecksum() {
+	try {
+		return execSync( 'git rev-parse --short HEAD' ).toString().replace( /\s/gm, '' );
+	} catch ( err ) {
+		return undefined;
+	}
+}
+
+export const BranchName = () => {
+	const branchName = getCurrentBranchName();
+	if ( branchName === 'master' ) {
+		return null;
+	}
+
+	return (
+		<span className={ 'environment branch-name' } title={ 'Commit ' + getCurrentCommitShortChecksum() }>
+			{ branchName }
+		</span>
+	);
+};
```

Comment:
I suspect the `<Branch />` component is also rendered client-side at some point, right? If that's the case, shouldn't we avoid calling `getCurrentBranchName()` and `getCurrentCommitShortChecksum()` on each render, since they are costly i/o operations?